File: mobile/src/main/java/jp/satorufujiwara/kotlin/ui/main/MainComponent.java
package jp.satorufujiwara.kotlin.ui.main;

import dagger.Component;
import jp.satorufujiwara.kotlin.AppComponent;
import jp.satorufujiwara.kotlin.DaggerApp;

@MainScope
@Component(
        dependencies = AppComponent.class,
        modules = MainModule.class
)
public interface MainComponent {

    void inject(MainActivity activity);

    void inject(MainFragment fragment);

    static class Initializer {

        private Initializer() {
        }

        public static MainComponent init(MainActivity activity) {
            return DaggerMainComponent.builder()
                    .appComponent(((DaggerApp) activity.getApplication()).appComponent())
                    .mainModule(new MainModule())
                    .build();
        }
    }
}

File: mobile/src/main/kotlin/jp/satorufujiwara/kotlin/ui/main/MainComponent.kt
package jp.satorufujiwara.kotlin.ui.main

import dagger.Component
import jp.satorufujiwara.kotlin.AppComponent
import jp.satorufujiwara.kotlin.DaggerApp

@MainScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(MainModule::class))
public interface MainComponent {

    public fun inject(activity: MainActivity)

    public fun inject(fragment: MainFragment)

    public object Initializer {

        public fun init(activity: MainActivity): MainComponent {
            return DaggerMainComponent.builder().appComponent(
                    (activity.application as DaggerApp).appComponent).mainModule(
                    MainModule()).build()
        }
    }
}
